kano_apps/AppData.py

```diff
@@ -105,21 +111,32 @@ def get_applications(parse_cmds=True):
                     if not is_app_installed(data):
                         data["_install"] = True
 
-                    apps.append(data)
+                    _applications[f] = data
                     if "overrides" in data:
-                        blacklist += data["overrides"]
+                        blacklist.extend(data["overrides"])
 
             if f[-8:] == ".desktop" and f[0:5] != "auto_":
                 data = _load_from_dentry(fp)
                 if data is not None:
-                    apps.append(data)
+                    _applications[f] = data
+
+        return _applications
 
-    filtered_apps = []
-    for app in apps:
-        if os.path.basename(app["origin"]) in blacklist:
-            continue
+    apps = _collect_apps(loc)
 
-        filtered_apps.append(app)
+    # get localised apps
+    current_locale, _ = locale.getlocale()
+    if current_locale:
+        locale_path = os.path.join(_SYSTEM_APPLICATIONS_LOC, "locale", current_locale)
+
+        localised_apps = _collect_apps(locale_path)
+        if localised_apps:
+            apps.update(localised_apps)
+
+    filtered_apps = [
+        app for f, app in apps.items()
```

Comment:
We are using python 2.7 so `.items()` returns a copy of a potentially large list - use `.iteritems()` instead.